File: Python/update_ticket.py
## This script requires "requests": http://docs.python-requests.org/
## To install: pip install requests

import requests
import json

api_key = "YOUR_API_KEY"
domain = "YOUR_DOMAIN"
password = "x"

# Id of the ticket to be updated
ticket_id = '57'

headers = { 'Content-Type' : 'application/json' }

ticket = {
  'subject' : 'Updated Title',
  'description' : 'Updated description',
  'priority' : 3,
}

r = requests.put("https://"+ domain +".freshdesk.com/api/v2/tickets/"+ticket_id, auth = (api_key, password), headers = headers, data = json.dumps(ticket))

if r.status_code == 200:
  print "Ticket updated successfully, the response is given below" + r.content
else:
  print "Failed to update ticket, errors are displayed below,"
  response = json.loads(r.content)
  print response["errors"]

  print "x-request-id : " + r.headers['x-request-id']
  print "Status Code : " + r.status_code

File: NodeJs/update_ticket.js
// Requires unirest module. To install:
// npm install unirest

var unirest = require('unirest');

var API_KEY = "API_KEY";
var FD_ENDPOINT = "YOUR_DOMAIN";

var PATH = "/api/v2/tickets";
var URL =  "https://" + FD_ENDPOINT + ".freshdesk.com"+ PATH + "/ticket_id";

var fields = {
  'subject': 'Ticket update subject',
  'description': 'Ticket update description.',
  'priority': 2,
  'status': 3
}

var Request = unirest.put(URL);

Request.auth({
  user: API_KEY,
  pass: "X",
  sendImmediately: true
})
.type('json')
.send(fields)
.end(function(response){
  	console.log(response.body)
    console.log("Response Status : " + response.status)
  	if(response.status != 200){
    	console.log("X-Request-Id :" + response.headers['x-request-id']);
    }
  });
